msal/application.py

```diff
@@ -1417,12 +1417,18 @@ def acquire_token_by_username_password(
             user_realm_result = self.authority.user_realm_discovery(
                 username, correlation_id=headers[msal.telemetry.CLIENT_REQUEST_ID])
             if user_realm_result.get("account_type") == "Federated":
-                response = _clean_up(self._acquire_token_by_username_password_federated(
-                    user_realm_result, username, password, scopes=scopes,
-                    data=data,
-                    headers=headers, **kwargs))
-                telemetry_context.update_telemetry(response)
-                return response
+                try:
+                    response = _clean_up(self._acquire_token_by_username_password_federated(
+                        user_realm_result, username, password, scopes=scopes,
+                        data=data,
+                        headers=headers, **kwargs))
+                except (ValueError, RuntimeError):
+                    raise RuntimeError(
+                        "ADFS is not configured properly. "
+                        "Consider use acquire_token_interactive() instead.")
```

Comment:
How about we include the original exception as an "inner" exception?

Reply:
Currently, the "inner" exception would show up in Python3's error trace already, therefore visible by default.

However, I would expect Azure CLI to do this, to have an actionable guidance most suitable in the Azure CLI's context:

```python
try:
    result = app.acquire_token_by_username_password(...)
except RuntimeError:
    raise CLIError('Username password login failed. You may consider using "az login" instead.')
```

Reply:
Won't this message be too vague? This exception chaining pattern is difficult for Azure CLI to get the inner exception message, like that of the original `(ValueError, RuntimeError)` and print it in the terminal.

Reply:
You can probably print the exception you caught. It would contain some (perhaps too much) info. Alas, it is hard to strike a balance here. :-(
```python
try:
    result = app.acquire_token_by_username_password(...)
except RuntimeError as e:
    raise CLIError('Username password login failed: {}. You may consider using "az login" instead.'.format(e))
```